src/Lazy/concurrent.ts

```diff
@@ -92,6 +92,10 @@ function concurrent<A>(
   if (length <= 0) {
     throw new RangeError("'length' must be over 0");
   }
+    
+  if (!length) {
+      throw new TypeError("'length' must be type of number");
+  }
```

Comment:
How about using `isNumber`?

If we use typescript, we can check the type on props.
but, Is it because of the case in the environment where javascript is used?

Reply:
Yes, that's right. As mentioned before, It was entirely my mistake to pass 'undefined' as the parameter.
It could have been easily prevented with TypeScript.

Upon further reflection, I'm not sure if my suggestion was necessary.

Thank you!